**Project 2: Biocomputing**

Name: Bhavya Sharma

Roll No.: 25210035

M.Tech: Biological Engineering

IIT Gandhinagar



**About the project-** This project, assigned as part of the BE623 (Biocomputing) course, involves the analysis of the Cytochrome P450 family of proteins, from sequence retrieval and filtering to structural prediction and validation. It utilizes Linux shell scripts and Python, utilising Jupyter Notebook run on Anaconda.


**Repository Structure**

Local repository- Project_2_Biocomputing

It contains the following folders/directories

data/- Contains all FASTA sequence files.

scripts/- Contains the shell scripts (bash and Python scripts) used to perform each analysis task.

results/- Contains all output files generated by the scripts.

README.txt- About project contents and other details.



Project_1_Biocomputing/

     data/   

             ten_sequences/
                          Arabidopsis_thaliana.fasta
                          Kitasatospora_aureofaciens.fasta
                          Pseudomonas_aeruginosa.fasta
                          Sphingopyxis_sp..fasta
                          Streptomyces_aurantiogriseus.fasta
                          Streptomyces_canus.fasta
                          Streptomyces_cyaneofuscatus.fasta
                          Streptomyces_glaucescens.fasta
                          Streptomyces_spiroverticillatus.fasta
                          Streptomyces_tendae.fasta           
             combined_10_sequences.fasta
             cytochrome_p450_sequences.fasta
             cytochrome_p450_sequences_filtered.fasta
     
     scripts/
             combine_10_fasta_files.sh
             download_p450.ipynb
             filter_short_sequences.sh
             Multiple_sequence_alignment.ipynb
             select_10_sequences.ipynb
             sequence_length_distribution.ipynb
             
      results/
             MSA_file.phylip
             MSA_Visualization.png
             Ramachandran_plot.pdf
             sequence_length_distribution.jpg
             sequences_aligned.fasta
             sequences_tree.dnd
             Structure_1_relaxed_rank_001_alphafold2.pdb
             Structure_2_relaxed_rank_001_alphafold2.pdb
             Structure1_visualization.png
             Structure2_visualization.png
      README.txt


**Steps Performed**

**Question 1:** Downloading Cytochrome P450 Protein Sequences
Used NCBI API via Biopython to download ~200 Cytochrome P450 sequences.
Script executed inside scripts/ folder (download_p450.ipynb).
Output saved as: data/cytochrome_p450_sequences.fasta

**Question 2:** Filtering Short Sequences (<100 aa)
Wrote a shell script filter_short_sequences.sh.
Removed sequences ≤100 aa to improve alignment quality.
Output stored as: cytochrome_p450_sequences_filtered.fasta

**Question 3:** Sequence Length Distribution
Computed and plotted a histogram of sequence lengths using matplotlib.
Plot saved in results/sequence_length_distribution.jpg.

**Question 4:** Extracting 1 sequence from 10 different organisms
Created the folder data/ten_sequences/.
Wrote a Python script to extract one representative sequence per organism.
Saved each as a separate FASTA file (10 total).

**Question 5:** Combining 10 FASTA Files
Wrote combine_10_fasta_files.sh.
Combined all individual FASTA files into:
data/combined_10_sequences.fasta

**Question 6:** Multiple Sequence Alignment (Clustal Omega)
Installed clustalo and performed MSA using Python API.
Output files:
results/sequences_aligned.fasta
results/sequences_tree.dnd

**Question 7:** MSA Visualization (Jalview)
Generated:
MSA_Visualization.png
MSA_file.phylip

**Question 8:** Phylogenetic Analysis (PhyML + SMS)
Used MSA_file.phylip for Maximum Likelihood tree.
Model chosen by Smart Model Selection:
Q.pfam + G + I + F
Generated:
Phylogram
Dendrogram

**Question 9:** Predicting 3D Structures Using AlphaFold2
Predicted structures for:
1. WYM67052.1 – Arabidopsis thaliana 
2. CAO3296666.1 – Sphingopyxis sp.
Generated:
Relaxed PDB files
pLDDT and PAE plots
Structural PNG visualizations (ChimeraX)

**Question 10:** Structure Validation Using PROCHECK
Validated Structure_1_relaxed_rank_001_alphafold2.pdb.
Ramachandran Plot generated
Plot saved as:
results/Ramachandran_plot.pdf


**Final Submission**
Git repository created on Linux.

All scripts, data, results, and analysis included.

README.txt prepared for documentation.

Pushed to GitHub


**Git Commands for Submission**

The following commands were used to set up the GitHub repository and submit the project.

1. Create a local repository named Project_2_Biocomputing
   
mkdir Project_2_Biocomputing

2. Change directory

cd Project_2_Biocomputing

3. Create the required subdirectories
   
mkdir data scripts results

4. Initialize a new Git repository
   
git init

5. Add all files to staging area
   
git add .   

6. Commit the files
   
git commit -m "Initial commit"

7. Add the remote GitHub repository
   
git remote add origin git@github.com:BhavyaSharma0903/Project_2_Biocomputing.git

8. Push to GitHub
   
git push -u origin master
